framework/src/modules/chain/blocks/utils.js

```diff
@@ -112,66 +31,44 @@ const parseStorageObjToLegacyObj = block => {
  * @returns {Object} cb.rows - List of blocks
  */
 // eslint-disable-next-line class-methods-use-this
-const loadBlocksDataWS = async (storage, filter, tx) => {
-	const params = { limit: filter.limit || 1 };
-
-	// FIXME: filter.id is not used
-	if (filter.id && filter.lastId) {
-		throw new Error('Invalid filter: Received both id and lastId');
+const loadBlocksFromLastBlockId = async (storage, lastBlockId, limit) => {
+	if (!lastBlockId) {
+		throw new Error('lastBlockId needs to be specified');
 	}
-	if (filter.id) {
-		params.id = filter.id;
-	} else if (filter.lastId) {
-		params.lastId = filter.lastId;
+	if (!limit) {
+		throw new Error('Limit needs to be specified');
 	}
 
 	// Get height of block with supplied ID
-	const rows = await storage.entities.Block.get(
-		{ id: filter.lastId || null },
-		{ limit: params.limit },
-		tx,
+	const [lastBlock] = await storage.entities.Block.get(
+		{ id: lastBlockId },
+		{ limit },
 	);
-	if (!rows.length) {
+	if (!lastBlock) {
 		throw new Error('Invalid lastBlockId requested');
```

Comment:
Would it make sense to also add the `lastBlockId` in the Error message? Can be beneficial info for debugging.